llvm/include/llvm/Support/PropertySetIO.h

```diff
@@ -0,0 +1,163 @@
+//==-- PropertySetIO.h -- models a sequence of property sets and their I/O -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// Models a sequence of property sets and their input and output operations.
+// TODO use Yaml as I/O engine.
```

Comment:
Such as https://llvm.org/docs/YamlIO.html ?
That sounds like a good idea.

Reply:
Yes, exactly.

Reply:
Just do it. :-)

Reply:
Yes,  in TODO list.